### ISSUE TITLE
How to programmatically get the number of current worker connections

### ISSUE THREAD

**Issue by andrii-korotkov-verkada** [benoitc/gunicorn#3152]
Hello,
I hope your day is going well. I'm looking to add some monitoring for services using gunicorn for the number of worker connections. This is to know when the workers are approaching the limit configured using --worker-connections flag and when it needs to be adjusted. I'm curious if there is some function like `get_current_number_of_gunicorn_connections()` I can call in a background thread publishing the metrics.
Thank you!

**Comment by andrii-korotkov-verkada** [comment]
Which metric is available for this?

**Comment by aamirawan7584** [comment]
I don't think so its directly available in Gunicorn but I think you can use pre_request and post-request hooks and maintain a global counter for number of connection which can be utilized later.

**Comment by mszeremet** [comment]
Server hooks are not reliable enough. Sometimes, when a worker terminates abnormally post_request hook doesn't work. Thus hook-based counter becomes inaccurate.

**Comment by benoitc** [comment]
Closing - prometheus support is planned and will be addressed in a future release.
